code/scripting/api/libs/mission.cpp

```diff
@@ -99,7 +99,35 @@ ADE_FUNC(evaluateSEXP, l_Mission, "string", "Runs the defined SEXP script", "boo
 		return ADE_RETURN_FALSE;
 }
 
-ADE_FUNC(runSEXP, l_Mission, "string", "Runs the defined SEXP script", "boolean", "if the operation was successful")
+ADE_FUNC(evalSEXP, l_Mission, "string", "Runs the defined SEXP script, and returns the result as a boolean", "boolean", "true if the SEXP returned SEXP_TRUE; false if the SEXP returned anything else (even a number)")
+{
+	const char* s;
+	int r_val;
+
+	if(!ade_get_args(L, "s", &s))
+		return ADE_RETURN_FALSE;
+
+	r_val = run_sexp(s);
+
+	if (r_val == SEXP_TRUE)
+		return ADE_RETURN_TRUE;
+	else
+		return ADE_RETURN_FALSE;
+}
+
+ADE_FUNC(evaluateNumericSEXP, l_Mission, "string", "Runs the defined SEXP script, and returns the result as a number", "number", "the value of the SEXP result")
+{
+	const char* s;
+	int r_val;
+
+	if (!ade_get_args(L, "s", &s))
+		return ade_set_args(L, "i", 0);
+
+	r_val = run_sexp(s);
```

Comment:
How would this handle the special NaN return values of the SEXP system? It may be worthwhile to consider those cases and return the float NaN in that case. I think I wrote code for that in the LuaSEXP system actually...

Reply:
Oh, that's a good point.  I'll add that.

Reply:
Ok, see what you think.